### PR TITLE
Single click to toggle default form flag

### PR DESCRIPTION
**Pull request by btry**
Add ability to change default flag with a single click from the forms list; make this flag available from the API

image

### CODE REVIEW

File: ajax/form.php [hunk 1]
```diff
@@ -30,14 +30,38 @@
  */
 
 include ('../../../inc/includes.php');
-Session::checkRight('entity', UPDATE);
```
Comment (btry) [inline ajax/form.php:33]: checkRight() outputs a HTML page on failure. This is not adapted with Ajax process.

File: ajax/form.php [hunk 2]
```diff
@@ -30,14 +30,38 @@
...
-   'id' => $_POST['id'],
-   'toggle' => 'toggle',
-]);
+if (!isset($_REQUEST['id']) || !isset($_REQUEST['action'])) {
+   http_response_code(400);
+   die();
+}
+
+$success = false;
+switch ($_REQUEST['action']) {
```
Comment (btry) [inline ajax/form.php:46]: group in a single entry point file the ajax calls related to forms for toggles.

File: inc/form.class.php [hunk 3]
```diff
@@ -288,6 +288,20 @@ public function rawSearchOptions() {
...
+         'field'              => 'is_default',
+         'datatype'           => 'specific',
+         'searchtype'         => [
+            '0'                  => 'equals',
+            '1'                  => 'notequals'
+         ],
+         'name'               => __('Default form in service catalog', 'formcreator'),
+         'searchtype'         => ['equals'],
+         'massiveaction'      => true
+      ];
```
Comment (btry) [inline inc/form.class.php:303]: add missing search option for this column

**Review by AdrienClairembault — APPROVED** [review]
Seems to be working fine.

**Pull request merged**